docs/octopus_deploy_smtp_config.md

```diff
@@ -0,0 +1,37 @@
+# octopus_deploy_smtp_config
+
+Tests the server-wide SMTP configuration. This can be found in the UI under Configuration -> SMTP
+
+## Example
+
+```ruby
+describe octopus_deploy_smtp_config do
```

Comment:
Dont i need to pass an api key and server url here?

Reply:
It's optional as of this set of changes (can go as ENV vars)

Reply:
(explained in the new readme and in `authentication.md`, which will come in a batch of docs later)